mode/javascript/javascript.js

```diff
@@ -617,6 +617,12 @@ CodeMirror.defineMode("javascript", function(config, parserConfig) {
   }
   function classBody(type, value) {
     if (type == "variable" || cx.style == "keyword") {
+      if (isTS) {
+        // Technically any of these modifiers could be standalone property names, but this is rare.
+        if (value == "public" || value == "private" || value == "protected" || value == "abstract") {
```

Comment:
Is a method like `public(...) {}` allowed in TS? (Similar to how you can do `get(...) {}` even though `get` is also the prefix for getters).